ietf/utils/tests/test_mega_menu.py

```diff
@@ -0,0 +1,108 @@
+from bs4 import BeautifulSoup
+import pytest
+from django.test import Client, RequestFactory
+
+from ietf.home.models import HomePage
+from ietf.standard.factories import StandardIndexPageFactory, StandardPageFactory
+from ietf.standard.models import StandardIndexPage, StandardPage
+from ietf.utils.models import MainMenuItem
+
+pytestmark = pytest.mark.django_db
+
+
+class TestMegaMenu:
+    @pytest.fixture(autouse=True)
+    def set_up(self, home: HomePage, client: Client):
+        self.home = home
+        self.client = client
+
+        self.standard_index: StandardIndexPage = StandardIndexPageFactory(
+            parent=self.home,
+        )  # type: ignore
+
+        self.standard_page: StandardPage = StandardPageFactory(
+            parent=self.standard_index,
+            show_in_menus=True,
+        )  # type: ignore
+
+    def test_primary_section(self):
```

Comment:
I'm a bit concerned that these tests would make it a bit difficult for FE devs to change the templates. (Unless only full-stack devs will ever be handling changes in the repo, in which case, you can probably ignore this.)

Reply:
> Unless only full-stack devs will ever be handling changes in the repo

You can never assume things like this :) But I can't think of another decent way to test the megamenu template. Sure, I can check the template context, but the template itself is quite complex.

On the other hand, the test is rather straightforward, perhaps a front-end dev could find their way around it: select something, check if it has the expected content. Python is quite readable.

Reply:
If we think these tests are needed and there's no other way around it, then I think the most helpful thing we could do is just add little bits of comments here and there on what each block of code is doing.